pkg/store/bucket.go

```diff
@@ -1610,6 +1631,9 @@ type postingPtr struct {
 // It returns one postings for each key, in the same order.
 // If postings for given key is not fetched, entry at given index will be nil.
 func (r *bucketIndexReader) fetchPostings(keys []labels.Label) ([]index.Postings, error) {
+	start := time.Now()
+	defer r.block.metrics.postingsLookupDuration.Observe(time.Since(start).Seconds())
```

Comment:
Isn't `time.Since(start).Seconds()` evaluated immediately? Should you do:

```
defer func() {
    r.block.metrics.postingsLookupDuration.Observe(time.Since(start).Seconds())
}
```

Reply:
You have x-ray eyes. Great catch 👍 
TIL there's a timer API https://github.com/prometheus/client_golang/blob/master/prometheus/timer.go#L20
I've fixed it using that.